addons/pos_mercado_pago/models/pos_payment_method.py

```diff
@@ -78,12 +67,18 @@ def mp_payment_intent_cancel(self, payment_intent_id):
         if not self.env.user.has_group('point_of_sale.group_pos_user'):
             raise AccessError(_("Do not have access to fetch token from Mercado Pago"))
 
-        mercado_pago = MercadoPagoPosRequest(self.sudo().mp_bearer_token)
+        mercado_pago = MercadoPagoPosRequest(self.sudo().pos_payment_provider_id.mp_bearer_token)
         # Call Mercado Pago for payment intend cancelation
         resp = mercado_pago.call_mercado_pago("delete", f"/point/integration-api/devices/{self.mp_id_point_smart_complet}/payment-intents/{payment_intent_id}", {})
         _logger.debug("mp_payment_intent_cancel(), response from Mercado Pago: %s", resp)
         return resp
 
+    def write(self, vals):
+        records = super().write(vals)
+        if 'pos_payment_provider_id' in vals and self.use_payment_terminal == 'mercado_pago':
+            self.mp_id_point_smart_complet = self._find_terminal(self.pos_payment_provider_id.mp_bearer_token, self.terminal_id)
```

Comment:
```suggestion
            self.mp_id_point_smart_complet = self._find_terminal(self.pos_payment_provider_id.mp_bearer_token, self.terminal_identifier)
```

I'm not sure if you forgot to adapt this with the terminal_identifier rename.